pkg/alertmanager/alertmanager_ring.go

```diff
@@ -32,6 +32,12 @@ var RingOp = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, func(s ring.InstanceS
 	return s != ring.ACTIVE
 })
 
+// UserOwnedRingOp is the operation used for checking if a user is owned by an alertmanager.
+var UserOwnedRingOp = ring.NewOp([]ring.InstanceState{ring.ACTIVE, ring.JOINING}, func(s ring.InstanceState) bool {
```

Comment:
Given the "is owned" logic is conceptually used in other places (eg. `ReadFullStateForUser()` also gets the set of alertmanagers owning a tenant) I'm wondering if we should call it `SyncRingOp` given this operation is specific to the sync operation done by the alertmanager itself.